jobs/async-upload/tests/test_models.py

```
@@ -0,0 +1,273 @@
+            artifact_id="any-value"  # Not validated for create_version
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_version
+        assert config.id == "test-model-id"
+        assert config.version_id == "test-version-id"
+
+    def test_create_version_intent_missing_model_id_fails(self):
+        """Test that create_version intent fails when model ID is missing"""
+        with pytest.raises(ValidationError) as exc_info:
+            ModelConfig(
+                upload_intent=UploadIntent.create_version,
+                id="",  # Empty string
+                version_id="test-version-id",
+                artifact_id="test-artifact-id"
+            )
+        
+        error_message = str(exc_info.value)
+        assert "Model ID must be set when intent is create_version" in error_message
+
+    def test_create_version_intent_without_version_id_succeeds(self):
+        """Test that create_version intent succeeds without version ID (only model ID required)"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_version,
+            id="test-model-id",
+            version_id="",  # Empty string - should be fine
+            artifact_id="test-artifact-id"
+        )
+        
+        # Should not raise any validation errors since only model ID is required
+        assert config.upload_intent == UploadIntent.create_version
+        assert config.id == "test-model-id"
+        assert config.version_id == ""
+        assert config.artifact_id == "test-artifact-id"
+
+    def test_create_version_intent_missing_model_id_only_fails(self):
+        """Test that create_version intent fails only when model ID is missing"""
+        with pytest.raises(ValidationError) as exc_info:
+            ModelConfig(
+                upload_intent=UploadIntent.create_version,
+                id="",  # Empty string - this should fail
```

that should fail for missing registered_model ID is fine ...